### ISSUE TITLE
[Question]: Adding proxy settings in the k8sgpt-operator

### ISSUE THREAD

**Issue by namgizlat** [k8sgpt-ai/k8sgpt#538]
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've included steps to reproduce the behavior

### Affected Components

- [X] K8sGPT (CLI)
- [X] K8sGPT Operator

### Steps to reproduce

1. Run k8sgpt analyze  --filter=Pod  --backend azureopenai --explain
2. Getting following timeout error:
Error: failed while calling AI provider azureopenai: Post "https://myopenapitests.azure-api.net/mtx/openai/deployments/gpt-35-turbo-002/chat/completions?api-version=2023-03-15-preview": dial tcp 22.201.29.165:443: i/o timeout

i am working behind proxy and added my proxy details in the deployment of k8sgpt-deployment.

### Expected behaviour

No timeout error and response from azureopenai

### Actual behaviour

following error:
Error: failed while calling AI provider azureopenai: Post "https://myopenapitests.azure-api.net/mtx/openai/deployments/gpt-35-turbo-002/chat/completions?api-version=2023-03-15-preview": dial tcp 22.201.29.165:443: i/o timeout

### Additional Information

i suspect the k8sgpt is trying to connect azure openai directly and not through the proxy

**Comment by arbreezy** [comment]
You are right @namgizlat , we don't currently support a proxy config for azure openai backend, it seems like a small change though, something close to
```
config := openai.DefaultConfig("token")
proxyUrl, err := url.Parse("http://{proxyAddress}:{port}")
if err != nil {
	panic(err)
}
transport := &http.Transport{
	Proxy: http.ProxyURL(proxyUrl),
}
config.HTTPClient = &http.Client{
	Transport: transport,
}

c := openai.NewClientWithConfig(config)
```
I will try to pick it up when I find spare time

**Comment by panneer-dev** [comment]
Thanks @arbreezy. Did you manage to find some spare time to take a look at proxy support ?
